frontend/src/components/management/LabworkSamples.tsx

```diff
@@ -0,0 +1,457 @@
+            Modal.warning({
+                title: "Warning",
+                content: `You cannot queue/dequeue more than ${MAX_SELECTION} samples.`,
+            })
+        } else {
+            setOpen(true)
+        }
+    }, [sampleSelectionCount])
+    const collapseRightPanel = useCallback(() => {
+        setOpen(false)
+    }, [])
+
+    return (
+        <>
+            <PagedItemsTable<ObjectWithSample>
+                getDataObjectsByID={mapSampleIDs}
+                pagedItems={samplesTableState}
+                columns={columns}
+                usingFilters={true}
+                initialLoad={false}
+                selection={selection}
+                topBarExtra={<Button onClick={maybeExpandRightPanel} disabled={sampleSelectionCount < 1}>{`Queue/Dequeue ${sampleSelectionCount} Samples`}</Button>}
+                paginationProps={{simple: true}}
+                {...samplesTableCallbacks}
+                setFilterCallback={setFilterCallback}
+                clearFiltersCallback={clearFiltersCallback}
+            />
+            <Drawer
+                title="Queue/Dequeue Samples"
+                placement="right"
+                size="large"
+                onClose={collapseRightPanel}
+                open={open}
+                destroyOnClose={true}
+            >
+                <LabworkSampleActions defaultSelection={defaultSelection} exceptedSampleIDs={exceptedSampleIDs} filters={wholeFilters} />
+            </Drawer>
+        </>
+    )
+}
+
+interface LabworkSampleActionsProps {
+    defaultSelection: boolean
+    exceptedSampleIDs: Sample['id'][]
+    filters: FilterSet
+}
+function LabworkSampleActions({ defaultSelection, exceptedSampleIDs, filters }: LabworkSampleActionsProps) {
+    const dispatch = useAppDispatch()
+
+    const [isFetching, setIsFetching] = useState(false)
+    interface ActionInfo {
+        study: Pick<FMSStudy, 'id' | 'letter'>
+        workflow: FMSWorkflow['name']
+        step: Step['name']
+        stepOrder: FMSSampleNextStepByStudy['step_order'],
+        alreadyQueued: Sample['id'][]
+    }
+    const [dequeueActions, setDequeueActions] = useState<ActionInfo[]>([])
+    const [queueActions, setQueueActions] = useState<ActionInfo[]>([])
+
+    interface StudyWorkflow {
+        study: Study['letter']
+        workflow: Workflow['name']
+    }
+    const [studyWorkflowsByProject, setStudyWorkflowsByProject] = useState<Record<Project['id'], StudyWorkflow[]>>({})
+
+    interface CommonProject {
+        id: Project['id']
+        name: Project['name']
+    }
+    const [commonProjects, setCommonProjects] = useState<CommonProject[]>([])
+
+    const [selectedProject, setSelectedProject] = useState<Project['id']>()
+    const [selectedStudyWorkflow, setSelectedStudyWorkflow] = useState<StudyWorkflow>()
+
+    const refreshActions = useCallback(async () => {
+        const sampleIDs = (await dispatch(api.samples.sample_ids_by_default_selection_and_excepted_ids(
```

Comment:
I am not convinced you need that endpoint ... We can ask for all samples matching a filter and exclude locally ... or we can add an exclude query param for the list endpoint. You are fetching the same sample from id the very next line ... The fetchSamples will get you the count too...

Reply:
Fair enough... I can check the exclude query param path.